### ISSUE TITLE
Initializing stateful properties for neighbor materials using coupled variables is not supported

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#21900]
## Reason
When using interface kernels the initialization of stateful materials will include a loop over neighbor materials. Any coupled variable in that loop will be uninitialized (size = 0), leading to a segfault.

## Design
Initialize coupled variables to the correct (neighbor?) values.

## Impact
Added capability.

Tagging @arovinelli @RLiu2020 @loganharbour

**Comment by dschwen** [comment]
Do neighbor materials have their own stateful properties? It looks like it at first glance., but that makes no sense to me. If I remove stateful neighbor material property initialization some test diff. That should not come as a surprise, as either

- the property is now uninitialized (if it has separate stateful storage)
- or the property is correctly initialized (and not clobbered over multiple times by the neighbor initialization)

**Comment by lindsayad** [comment]
Not quite sure why I wasn't tagged here. I don't think this should be very hard to fix. I bet I could take care of it next week unless someone else has started in on it

**Comment by lindsayad** [comment]
well I shouldn't overpromise ... I bet I have some reports I need to focus on next week

**Comment by hugary1995** [comment]
Oh perhaps you came here from that discussion? Anyways...

**Comment by lindsayad** [comment]
yes I did. Got here due to your reference of the issue in that discussion, so I appreciate you dropping that there!
